controllers/packetmachine_controller.go

```diff
@@ -412,6 +412,16 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 		result = ctrl.Result{}
 	}
 
+	// If Metro or Facility has changed in the spec, verify that the facility's metro is compatible with the requested spec change.
+
+	if machineScope.PacketMachine.Spec.Facility != "" && machineScope.PacketMachine.Spec.Facility != dev.Facility.Code {
+		return ctrl.Result{}, fmt.Errorf("instance facility does not match machine facility %s: %q != %q", machineScope.Name(), machineScope.PacketMachine.Spec.Facility, dev.Facility.Code)
+	}
+
+	if machineScope.PacketMachine.Spec.Metro != "" && machineScope.PacketMachine.Spec.Metro != dev.Metro.Code {
+		return ctrl.Result{}, fmt.Errorf("instance metro does not match machine metro %s: %w", machineScope.Name(), err)
```

Comment:
Where is `err` coming from here?

Reply:
we've added a new error string for this case as well now.